plugins/sign-on/openid-connect-generic-client/includes/openid-connect-generic-client-wrapper.php

```diff
@@ -465,6 +466,34 @@ function authentication_request_callback() {
 		exit;
 	}
 
+	function get_common_login($user_claim) {
+		if ( $user_claim['https://login.bcc.no/claims/churchName'] == get_option('bcc_local_church') ) {
+			$role = array(
+				'title' => 'BCC Local',
+				'id' => 'bcc_local'
+			);
+		} else {
+			$role = array(
```

Comment:
Is "IsMember" getting checked anywhere, or are we just assuming that all logged in users are members?

Reply:
Yes. Members with no active membership will not pass the Auth0 login.

Reply:
Ok, I would add it to the plugin as well in that case. There is a claim for membership, and the way Auth0 login works is beyond our control. It would make perfect sense to allow some non-members to log in with Auth0 in some cases.

Reply:
I'm not sure I understand this.

Under **Authorization Settings** in the **OpenID Connect - Generic Client** settings we have **Enforce Privacy** which, as far as I understand, if it's set to true then only active members should have access to visit the website.

As well, I see we have a custom rule set up in Auth0 which checks for the membership. To be honest, I don't think that should be the plugin's decision to allow non-members to log in.

Also, if I add some code in the `get_common_login `function, what should I add?

Reply:
I would just add a check that the user has the IsMember claim. If not, they can be treated as an anonymous user.

Reply:
Ok.